main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1908,9 +1908,10 @@ object Weeder2 {
       flatMapN(pickNameIdent(tree), exprs) {
         case (ident, channel :: body :: Nil) =>
           Validation.success(SelectChannelRule(ident, channel, body))
-        case _ =>
+        case _ => // Unreachable case, since this is a parser error
           val err = Malformed(NamedTokenSet.MatchRule, SyntacticContext.Expr.OtherExpr, loc = tree.loc)
-          Validation.HardFailure(Chain(err))
+          val ident = Name.Ident("error", tree.loc.asSynthetic)
+          Validation.toSoftFailure(SelectChannelRule(ident, Expr.Error(err), Expr.Error(err)), err)
```

Comment:
we could also throw an ICE but my hope is that if we decide to change parsing at some point then this will catch it

Reply:
Ditto for comment about ICE